File: backend/app/Models/BoostPlan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

#[Fillable(['code', 'name', 'price', 'duration_hours', 'is_active'])]
class BoostPlan extends Model
{
    use HasFactory;

    public const CODE_72H = '72h';

    public const CODE_7D = '7d';

    public const CODE_30D = '30d';

    protected function casts(): array
    {
        return [
            'price' => 'integer',
            'duration_hours' => 'integer',
            'is_active' => 'boolean',
        ];
    }

    public function boosts(): HasMany
    {
        return $this->hasMany(Boost::class);
    }
}

File: mobile/lib/models/boost_plan.dart
class BoostPlan {
  const BoostPlan({
    required this.id,
    required this.code,
    required this.name,
    required this.price,
    required this.durationHours,
  });

  final int id;
  final String code;
  final String name;
  final int price;
  final int durationHours;

  factory BoostPlan.fromJson(Map<String, dynamic> json) => BoostPlan(
    id: json['id'] as int,
    code: json['code'] as String,
    name: json['name'] as String,
    price: json['price'] as int,
    durationHours: json['duration_hours'] as int,
  );
}
